candy-machine/program/src/processor/mint.rs

```diff
@@ -608,6 +622,152 @@ pub fn handle_mint_nft<'info>(
         &[&authority_seeds],
     )?;
 
+    if let Some(lockup_settings) = &candy_machine.data.lockup_settings {
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            punish_bots(
+                CandyError::LockupSettingsMissingAccounts,
+                payer.to_account_info(),
+                ctx.accounts.candy_machine.to_account_info(),
+                ctx.accounts.system_program.to_account_info(),
+                BOT_FEE,
+            )?;
+            return Ok(());
+        }
+
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            return err!(CandyError::LockupSettingsMissingTokenManager);
+        }
+        let token_manager = &ctx.remaining_accounts[remaining_accounts_counter];
+
+        remaining_accounts_counter += 1;
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            return err!(CandyError::LockupSettingsMissingTokenManagerTokenAccount);
+        }
+        let token_manager_token_account = &ctx.remaining_accounts[remaining_accounts_counter];
+
+        remaining_accounts_counter += 1;
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            return err!(CandyError::LockupSettingsMissingMintCounter);
+        }
+        let mint_counter = &ctx.remaining_accounts[remaining_accounts_counter];
+
+        remaining_accounts_counter += 1;
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            return err!(CandyError::LockupSettingsMissingRecipientTokenAccount);
+        }
+        let recipient_token_account = &ctx.remaining_accounts[remaining_accounts_counter];
+
+        remaining_accounts_counter += 1;
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            return err!(CandyError::LockupSettingsMissingTimeInvalidator);
+        }
+        let time_invalidator = &ctx.remaining_accounts[remaining_accounts_counter];
+
+        remaining_accounts_counter += 1;
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            return err!(CandyError::LockupSettingsMissingTimeInvalidatorProgram);
+        }
+        let time_invalidator_program = &ctx.remaining_accounts[remaining_accounts_counter];
+
+        remaining_accounts_counter += 1;
+        if ctx.remaining_accounts.len() <= remaining_accounts_counter {
+            return err!(CandyError::LockupSettingsMissingTokenManagerProgram);
+        }
+        let token_manager_program = &ctx.remaining_accounts[remaining_accounts_counter];
+        // If we add more extra accounts later on we need to uncomment the following line out.
+        // remaining_accounts_counter += 1;
```

Comment:
not sure why this was commented out or what the harm is for just adding + one to avoid confusion for future additions, but it was like this above so we left it

Reply:
Commented out because of the dead code lint

Reply:
ah makes sense, in this case the linting doesn't complain just leaving it as a += 1 maybe because its a &mut
https://github.com/metaplex-foundation/metaplex-program-library/pull/511/files#diff-a5b4d95737a7474da174757a8ff945dad5df3db8b0d6b0f8feecf4c91d2bca44R908